rs/replicated_state/src/canister_state/queues.rs

```diff
@@ -75,62 +74,13 @@ pub struct CanisterQueues {
     /// Queue of ingress (user) messages.
     ingress_queue: IngressQueue,
 
-    /// Per remote canister input and output queues.
-    canister_queues: BTreeMap<CanisterId, (InputQueue, OutputQueue)>,
-
-    /// FIFO queue of local subnet sender canister IDs ensuring round-robin
-    /// consumption of input messages. Only senders with non-empty queues
-    /// are scheduled.
-    ///
-    /// We rely on `ReplicatedState::canister_states` to decide whether a canister
-    /// is local or not. This test is subject to race conditions (e.g. if the sender
-    /// has just been deleted), meaning that the separation into local and remote
-    /// senders is best effort.
-    local_subnet_input_schedule: VecDeque<CanisterId>,
-
-    /// FIFO queue of remote subnet sender canister IDs ensuring round-robin
-    /// consumption of input messages. Only senders with non-empty queues
-    /// are scheduled.
-    ///
-    /// We rely on `ReplicatedState::canister_states` to decide whether a canister
-    /// is local or not. This test is subject to race conditions (e.g. if the sender
-    /// has just been deleted), meaning that the separation into local and remote
-    /// senders is best effort.
-    remote_subnet_input_schedule: VecDeque<CanisterId>,
-
-    /// Running `input_queues` stats.
-    input_queues_stats: InputQueuesStats,
-
-    /// Running `output_queues` stats.
-    output_queues_stats: OutputQueuesStats,
-
-    /// Running memory usage stats, across input and output queues.
-    memory_usage_stats: MemoryUsageStats,
-
-    /// Round-robin across ingress and cross-net input queues for pop_input().
-    next_input_queue: NextInputQueue,
-}
-
-/// Wrapper around the induction pool (ingress and input queues); a priority
-/// queue used for round-robin scheduling of senders when consuming input
-/// messages; and output queues.
-///
-/// Responsible for queue lifetime management, fair scheduling of inputs across
-/// sender canisters and queue backpressure.
-///
-/// Encapsulates the `InductionPool` component described in the spec. The reason
-/// for bundling together the induction pool and output queues is to reliably
-/// implement backpressure via queue slot reservations for response messages.
-#[derive(Clone, Debug, Default, PartialEq, Eq)]
-pub struct NewCanisterQueues {
-    /// Queue of ingress (user) messages.
-    ingress_queue: IngressQueue,
-
-    /// Per remote canister input and output queues.
+    /// Per remote canister input and output queues. Queues hold references into the
+    /// message pool, some of which may be stale due to expiration or load shedding.
+    /// The item at the head of each queue, if any, is guaranteed tp be non-stale.
```

Comment:
How about introducing a function `assert_invariants` (or maybe even `enforce_invariants_and_debug_assert_that_it_was_not_necessary`) and calling it at the end of each relevant function? This would also document the invariants clearly.